src/redisearch_api.c

```diff
@@ -106,21 +108,23 @@ RSField* RediSearch_CreateField(IndexSpec* sp, const char* name, unsigned types,
   }
 
   RWLOCK_RELEASE();
-
-  return fs;
+  return fs->index;
 }
 
-void RediSearch_TextFieldSetWeight(IndexSpec* sp, FieldSpec* fs, double w) {
+void RediSearch_TextFieldSetWeight(IndexSpec* sp, RSFieldID id, double w) {
+  FieldSpec* fs = sp->fields + id;
   assert(FIELD_IS(fs, INDEXFLD_T_FULLTEXT));
   fs->ftWeight = w;
 }
 
-void RediSearch_TagSetSeparator(FieldSpec* fs, char sep) {
+void RediSearch_TagSetSeparator(IndexSpec* sp, RSFieldID id, char sep) {
+  FieldSpec* fs = sp->fields + id;
   assert(FIELD_IS(fs, INDEXFLD_T_TAG));
   fs->tagSep = sep;
 }
 
-void RediSearch_TagCaseSensitive(FieldSpec* fs, int enable) {
+void RediSearch_TagFieldSetCaseSensitive(IndexSpec* sp, RSFieldID id, int enable) {
+  FieldSpec* fs = sp->fields + id;
```

Comment:
@mnunberg I think the better option is what I suggested here:
https://github.com/RediSearch/RediSearch/issues/823

Reply:
@MeirShpilraien as I commented there, I think it will provide weird semantics in requiring you to "free" the field.. It would be possible to keep a list of field struct-pointers, but this is also  unreasonable. We should make the API functions consistent anyway, accepting both a spec and a field..